System/Posix/Files/Common.hsc

```
@@ -688,3 +752,398 @@ getFdPathVar (Fd fd) v =
+#else
+pattern StatxMode = StatxMask 0
+#endif
+
+pattern StatxNlink :: StatxMask
+#ifdef STATX_NLINK
+pattern StatxNlink = StatxMask (#const STATX_NLINK)
+#else
+pattern StatxNlink = StatxMask 0
+#endif
+
+pattern StatxUid :: StatxMask
+#ifdef STATX_UID
+pattern StatxUid = StatxMask (#const STATX_UID)
+#else
+pattern StatxUid = StatxMask 0
+#endif
+
+pattern StatxGid :: StatxMask
+#ifdef STATX_GID
+pattern StatxGid = StatxMask (#const STATX_GID)
+#else
+pattern StatxGid = StatxMask 0
+#endif
+
+pattern StatxAtime :: StatxMask
+#ifdef STATX_ATIME
+pattern StatxAtime = StatxMask (#const STATX_ATIME)
+#else
+pattern StatxAtime = StatxMask 0
+#endif
+
+pattern StatxMtime :: StatxMask
+#ifdef STATX_MTIME
+pattern StatxMtime = StatxMask (#const STATX_MTIME)
+#else
+pattern StatxMtime = StatxMask 0
+#endif
+
+pattern StatxCtime :: StatxMask
+#ifdef STATX_CTIME
+pattern StatxCtime = StatxMask (#const STATX_CTIME)
+#else
+pattern StatxCtime = StatxMask 0
+#endif
+
+pattern StatxBtime :: StatxMask
+#ifdef STATX_BTIME
+pattern StatxBtime = StatxMask (#const STATX_BTIME)
+#else
+pattern StatxBtime = StatxMask 0
+#endif
+
+pattern StatxIno :: StatxMask
+#ifdef STATX_INO
+pattern StatxIno = StatxMask (#const STATX_INO)
+#else
+pattern StatxIno = StatxMask 0
+#endif
+
+pattern StatxSize :: StatxMask
+#ifdef STATX_SIZE
+pattern StatxSize = StatxMask (#const STATX_SIZE)
+#else
+pattern StatxSize = StatxMask 0
+#endif
+
+pattern StatxBlocks :: StatxMask
+#ifdef STATX_BLOCKS
+pattern StatxBlocks = StatxMask (#const STATX_BLOCKS)
+#else
+pattern StatxBlocks = StatxMask 0
+#endif
+
+pattern StatxBasicStats :: StatxMask
+#ifdef STATX_BASIC_STATS
+pattern StatxBasicStats = StatxMask (#const STATX_BASIC_STATS)
+#else
+pattern StatxBasicStats = StatxMask 0
+#endif
+
+pattern StatxAll :: StatxMask
+#ifdef STATX_ALL
+pattern StatxAll = StatxMask (#const STATX_ALL)
+#else
+pattern StatxAll = StatxMask 0
+#endif
+
+
+defaultStatxMask :: StatxMask
+defaultStatxMask = StatxAll
```

I set the default mask to `STATX_ALL` here.